components/sup/src/main.rs

```diff
@@ -851,26 +852,39 @@ fn sub_status(m: &ArgMatches) -> Result<()> {
                     process::exit(2);
                 }
             };
-
-            for spec in specs {
-                let status = Manager::service_status(&cfg, &spec.ident)?;
-                outputln!("{}", status);
-            }
+            print_statuses(specs.iter()
+                                .filter_map(|spec| Manager::service_status(&cfg, &spec.ident).ok())
+                                .collect::<Vec<ServiceStatus>>())?;
         }
         None => {
-            let statuses = Manager::status(&cfg)?;
-            if statuses.is_empty() {
-                println!("No services loaded.");
-                return Ok(());
-            }
-            for status in statuses {
-                println!("{}", status);
-            }
+            print_statuses(Manager::status(&cfg)?)?;
         }
     }
     Ok(())
 }
 
+fn print_statuses(statuses: Vec<ServiceStatus>) -> Result<()> {
+    if statuses.is_empty() {
+        println!("No services loaded.");
+        return Ok(());
+    }
+    let mut tw = cli::TableWriter::new(7);
+    tw.set_titles(vec!("package", "type", "state", "uptime", "pid", "group", "style"));
+    for status in statuses {
+        tw.add_line(vec!(
+            status.pkg.ident.to_string(),
+            status.composite.unwrap_or("standalone".to_string()),
+            status.process.state.to_string(),
+            status.process.elapsed.num_seconds().to_string() + "s",
```

Comment:
It may be easier for any downstream Unix pipeline commands to parse the data if we exclude the "s" suffix here. 

We might want to incorporate that into the column header instead. What are the separator characters here, tabs or spaces? I wouldn't want a header like `"uptime (sec)"` to throw off the column alignment due to the space.

Reply:
Ok I can move it. Makes sense. The separator for columns is `\t` so we're good.